examples/tensorflow/train.py

```diff
@@ -0,0 +1,160 @@
+# tensorflow 2.x core api
+import mlflow
+import pandas as pd
+
+import tensorflow as tf
+from sklearn.datasets import fetch_california_housing
+
+
+class Normalize(tf.Module):
+    """Data Normalization class"""
+
+    def __init__(self, x):
+        # Initialize the mean and standard deviation for normalization
+        self.mean = tf.math.reduce_mean(x, axis=0)
+        self.std = tf.math.reduce_std(x, axis=0)
+
+    def norm(self, x):
+        return (x - self.mean) / self.std
+
+    def unnorm(self, x):
+        return (x * self.std) + self.mean
+
+
+class LinearRegression(tf.Module):
+    """Linear Regression model class"""
+
+    def __init__(self):
+        self.built = False
+
+    @tf.function
+    def __call__(self, x):
+        # Initialize the model parameters on the first call
+        if not self.built:
+            # Randomly generate the weight vector and bias term
+            rand_w = tf.random.uniform(shape=[x.shape[-1], 1])
+            rand_b = tf.random.uniform(shape=[])
+            self.w = tf.Variable(rand_w)
+            self.b = tf.Variable(rand_b)
+            self.built = True
+        y = tf.add(tf.matmul(x, self.w), self.b)
+        return tf.squeeze(y, axis=1)
+
+
+class ExportModule(tf.Module):
+    """Exporting TF model"""
+
+    def __init__(self, model, norm_x, norm_y):
+        # Initialize pre and postprocessing functions
+        self.model = model
+        self.norm_x = norm_x
+        self.norm_y = norm_y
+
+    @tf.function(input_signature=[tf.TensorSpec(shape=[None, None], dtype=tf.float32)])
+    def __call__(self, x):
+        # Run the ExportModule for new data points
+        x = self.norm_x.norm(x)
+        y = self.model(x)
+        y = self.norm_y.unnorm(y)
+        return y
+
+
+def mse_loss(y_pred, y):
+    """Calculating Mean Square Error Loss function"""
+    return tf.reduce_mean(tf.square(y_pred - y))
+
+
+if __name__ == "__main__":
+    # Set a random seed for reproducible results
+    tf.random.set_seed(42)
+
+    # Load dataset
+    dataset = pd.DataFrame(data=fetch_california_housing(as_frame=True)["frame"])
```

Comment:
```suggestion
    dataset = fetch_california_housing(as_frame=True)["frame"]
```

Do we need `pd.DataFrame`?

```python
>>> type(fetch_california_housing(as_frame=True)["frame"])
<class 'pandas.core.frame.DataFrame'>
```